apollo-compiler/build.gradle.kts

```diff
@@ -25,27 +25,38 @@ dependencies {
   add("testImplementation", groovy.util.Eval.x(project, "x.dep.truth"))
 }
 
-tasks.register("pluginVersion") {
-  val outputDir = file("src/generated/kotlin")
+abstract class GeneratePluginVersion : DefaultTask() {
+  @get:org.gradle.api.tasks.Input
+  abstract val version: Property<String>
 
-  inputs.property("version", version)
-  outputs.dir(outputDir)
+  @get:org.gradle.api.tasks.OutputFile
+  abstract val outputFile: RegularFileProperty
 
-  doLast {
-    val versionFile = file("$outputDir/com/apollographql/android/Version.kt")
+  @org.gradle.api.tasks.TaskAction
+  fun taskAction() {
+    val versionFile = outputFile.asFile.get()
     versionFile.parentFile.mkdirs()
     versionFile.writeText("""// Generated file. Do not edit!
-package com.apollographql.android
+package com.apollographql.apollo.compiler
 val VERSION = "${project.version}"
 """)
   }
 }
 
-tasks.getByName("compileKotlin").dependsOn("pluginVersion")
```

Comment:
Looks like this wasn't working before since the generated `Versions.kt` file was not added to the list of sources.